### ISSUE TITLE
Soundness bug in Resource Leak Checker for type wrapping multiple resources

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#5453]
Example:
```java
@InheritableMustCall({"finish1", "finish2"})
 class TwoOwningMCATest {
   @Owning private final Foo f1 = new Foo();
   @Owning private final Foo f2;
   @MustCallAlias
   TwoOwningMCATest(@MustCallAlias Foo g) {
     this.f2 = g;
   }
   @EnsuresCalledMethods(value="this.f1", methods="a")
   void finish1() {
     this.f1.a();
   }
   @EnsuresCalledMethods(value="this.f2", methods="a")
   void finish2() {
     this.f2.a();
   }
   @InheritableMustCall("a")
   static class Foo {
     void a() {}
   }
 }
```
The Resource Leak Checker reports no error for the above.  But the `@MustCallAlias` annotations on the constructor are invalid, as satisfying the must-call obligation of the parameter is _not_ sufficient to satisfy the obligations of the newly-created `TwoOwningMCATest` object.

/cc @kelloggm @Nargeshdb
